src/ILVerification/src/TypeVerifier.cs

```diff
@@ -98,36 +99,59 @@ public void VerifyInterfaces()
                         MethodDesc resolvedMethod = type.ResolveInterfaceMethodTarget(method);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType, implementedInterface.InterfaceImplementation), Format(method));
                         }
                     }
                 }
             }
         }
 
-        // Format helpers for future use
-        // The idea is to improve formatting string with tokens value i.e.: 
-        // [Assembly]Class(0x1234) vs current [Assembly]Class
-        // in case somebody will pass '--tokens' switch to ILVerify
         private string Format(EcmaType type)
         {
-            return type.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return string.Format("{0}([{1}]0x{2:X8})", type, _module, _module.MetadataReader.GetToken(_typeDefinitionHandle));
+            }
+            else
+            {
+                return type.ToString();
+            }
         }
 
-        private string Format(DefType interfaceTypeDef)
+        private string Format(DefType defType, InterfaceImplementation interfaceImplementation)
         {
-            return interfaceTypeDef.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return string.Format("{0}([{1}]0x{2:X8})", defType, _module, _module.MetadataReader.GetToken(interfaceImplementation.Interface));
+            }
+            else
+            {
+                return defType.ToString();
+            }
         }
 
         private string Format(MethodDesc methodDesc)
         {
-            return methodDesc.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                TypeDesc typeDesc = methodDesc.OwningType.GetTypeDefinition();
+                EcmaAssembly ecmaAssembly = (EcmaAssembly)((MetadataType)typeDesc).Module;
+                ILVerifyTypeSystemContext typeSystemContext = (ILVerifyTypeSystemContext)ecmaAssembly.Context;
```

Comment:
You do not need to go through the lookup in the Context. I think this can be just:
```csharp
TypeDesc typeDesc = methodDesc.OwningType.GetTypeDefinition();
EcmaModule module = (EcmaModule)((MetadataType)typeDesc).Module;
MetadataReader reader = module.MetadataReader;
```

Reply:
right, missed EcmaAssemlby : EcmaModule